backend/src/main/java/com/google/sps/FindMatchQuery.java

```diff
@@ -14,42 +14,32 @@
 
 package com.google.sps;
 
-import com.google.gson.Gson;
 import com.google.sps.data.Match;
 import com.google.sps.data.Participant;
-import java.time.Clock;
-import java.time.Duration;
-import java.time.Instant;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Date;
-import java.util.Collection;
-import java.util.Collections;
+import java.time.ZonedDateTime;
 import java.util.List;
-import java.util.stream.Collectors;
 
 /** Class used to find a match in a list of Participants with the most recently added Participant */
 public final class FindMatchQuery {
 
-  private Clock clock;
-  private static int MAX_DURATION_DIFF = 15; // maximum difference in duration to be compatible
-  private static int PADDING_TIME = 15; // extra padding time to ensure large enough meeting time block
-  
-  /** Constructor */
-  public FindMatchQuery() {
-    clock = new Clock();
-  }
+  /** Reference date time for "current" time */
+  private ZonedDateTime dateTime;
+  /** Maximum difference in duration to be compatible */
+  private static int MAX_DURATION_DIFF = 15;
+  /** extra padding time to ensure large enough meeting time block */
+  private static int PADDING_TIME = 15;
 
-  /** Constructor with manually set date */
-  public FindMatchQuery(Clock clock) {
-    this.clock = clock;
+  /** Set "current" date and time to calculate whether a match is possible in time */
```

Comment:
Sorry to keep harping on this but I still think we can find a better name for the field and parameter here. "Current" time doesn't work because it will be stale immediately after instantiation. For example if this class were instantiated, kept alive for 24 hours, and then used to perform a match, it would produce incorrect results because this field is inaccurate. 

The right approach here would be to get the current time `ZoneDateTime` within the findMatch method so that it's always accurate. You can use ZoneDateTime.now(Clock) to get the current time within `findMatch` while also constructing this class with a Clock (as before) to facilitate testing.

Reply:
Since I instantiate this class every time a new participant is added, the "current" time is the time the participant is added. Should I make the FindMeetingQuery class static so I don't have to instantiate it every time? Or should I just keep it the way it is now and do the ZonedDateTime.now(Clock) thing?

Reply:
Clock sounds like the right way to go. The code instantiates the class every time a new participant is added 'now' but it's possible someone could come along later and try to reuse the class multiple times even though currently it only works correctly immediately after instantiation. Changing to a clock would allow using a single instance for multiple runs of findMatch.

Reply:
Ah I see, thanks for explaining. Is it correct now? I used a fixed clock in the match query test, and in AddParticipantServlet, I passed Clock.systemUTC() to FindMatchQuery(), and instantiated ZonedDateTime in the findMatchQuery() method.